dds/idl/marshal_generator.cpp

```diff
@@ -2460,21 +2458,48 @@ bool marshal_generator::gen_struct(AST_Structure* node,
     } else {
       string expr;
       for (size_t i = 0; i < fields.size(); ++i) {
-        if (i) expr += "\n    && ";
+        if (i && exten != extensibilitykind_appendable) {
+          expr += "\n    && ";
+        }
         const string field_name = fields[i]->local_name()->get_string();
         const string cond = rtpsCustom.getConditional(field_name);
         if (!cond.empty()) {
           expr += rtpsCustom.preFieldRead(field_name);
           expr += "(!(" + cond + ") || ";
         }
+        // TODO (sonndinh): Integrate with try-construct for when the stream
+        // ends before some fields on the reader side get their values.
+        if (exten == extensibilitykind_appendable) {
+          expr += "  if (strm.encoding().xcdr_version() == Encoding::XCDR_VERSION_2) {\n";
```

Comment:
This if statement looks like it would be the middle of a expression for the special RTPS generation, which matches up with what the PR builder error is saying:

> RtpsCoreTypeSupportImpl.cpp:12328:34: error: expected primary-expression before 'if'